feat(to_unixtime): add timestamp types as arguments

etolbakov opened the pull request:
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Add more argument types for `to_unixtime` function

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [X]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link
https://github.com/GreptimeTeam/greptimedb/issues/1103

killme2008 commented:
Is it still working? We can ship it in 0.3.1 next week.

etolbakov commented:
Hey Dennis, 
yes, I was catching up with @waynexia a couple of days ago regarding the question that I had.
I'm clear now and will try allocating some time during the weekend.

etolbakov commented on src/common/function/src/scalars/timestamp/to_unixtime.rs:
```diff
@@ -79,6 +91,43 @@ impl Function for ToUnixtimeFunction {
+            ConcreteDataType::Timestamp(ts) => {
+                let array = columns[0].to_arrow_array();
+                let value = match ts {
+                    TimestampType::Second(_) => {
```
a bit of duplication here..was trying to extract a generic method but could not find one so far. Your suggestions will be highly appreciated 🙏

etolbakov commented:
@killme2008 @waynexia 
Could you please take a look when you have a spare minute?

killme2008 reviewed (approved):
LGTM

waynexia commented on tests/cases/standalone/common/select/dummy.sql:
```diff
@@ -11,3 +11,21 @@ select "A";
+
+select TO_UNIXTIME(b) from test_unixtime;
+
+DROP TABLE test_unixtime;
```
nit: lack of empty line

waynexia reviewed (approved):
LGTM, thanks @etolbakov 👍

etolbakov commented:
@killme2008 @waynexia
I've addressed code review suggestions.
Thanks for pointing me out the `paste` crate, I found it useful.
I see that the `test_alter_region_with_reopen` test failed, I've double checked it works locally fine. Could it be the flaky behaviour? Would you mind just restart the action for me please?

killme2008 commented:
> @killme2008 @waynexia I've addressed code review suggestions. Thanks for pointing me out the `paste` crate, I found it useful. I see that the `test_alter_region_with_reopen` test failed, I've double checked it works locally fine. Could it be the flaky behaviour? Would you mind just restart the action for me please?

Great work. Looks like it's a known bug, will be fixed at https://github.com/GreptimeTeam/greptimedb/pull/1759
